File: src/main/java/com/hedvig/claims/query/ClaimReportEntity.java
package com.hedvig.claims.query;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.Year;

@Data
@AllArgsConstructor
@Entity
public class ClaimReportEntity {
  @Id
  private String claimId;
  private String memberId;
  private BigDecimal grossPaid;
  private BigDecimal reserved;
  private LocalDate dateOfLoss;
  private LocalDate notificationDate;
  private Integer claimYear;
  private String descriptionOfLoss;
  private String currency;
  private String claimStatus;
  private LocalDate claimStatusLastUpdated;

  public ClaimReportEntity(String claimId, String memberId, LocalDate notificationDate, String claimStatus, LocalDate claimStatusLastUpdated) {
    this.claimId = claimId;
    this.memberId = memberId;
    this.notificationDate = notificationDate;
    this.claimStatus = claimStatus;
    this.claimStatusLastUpdated = claimStatusLastUpdated;
  }
}

File: src/main/java/com/hedvig/claims/query/ClaimReportEntity.kt
package com.hedvig.claims.query

import java.math.BigDecimal
import java.time.LocalDate
import java.time.Year
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class ClaimReportEntity(
    @Id
    val claimId: String,
    val memberId: String,
    var grossPaid: BigDecimal?,
    var reserved: BigDecimal?,
    var totalIncurred: BigDecimal?
) {

    lateinit var dateOfLoss: LocalDate
    lateinit var notificationDate: LocalDate
    lateinit var claimYear: Year
    lateinit var descriptionOfLoss: String
    lateinit var currency: String
    lateinit var claimStatus: String
    lateinit var claimStatusLastUpdated: LocalDate

    constructor(
        claimId: String,
        memberId: String,
        notificationDate: LocalDate,
        claimStatus: String,
        claimStatusLastUpdated: LocalDate
    ) :
            this(
                claimId,
                memberId,
                null,
                null,
                null
            ) {
        this.notificationDate = notificationDate
        this.claimStatus = claimStatus
        this.claimStatusLastUpdated = claimStatusLastUpdated
    }

}
